Source/Diagnostics/ComputeDiagFunctors/BackTransformFunctor.cpp

```diff
@@ -112,14 +114,33 @@ BackTransformFunctor::operator ()(amrex::MultiFab& mf_dst, int /*dcomp*/, const
             const Box& tbx = mfi.tilebox();
             amrex::Array4<amrex::Real> src_arr = tmp[mfi].array();
             amrex::Array4<amrex::Real> dst_arr = mf_dst[mfi].array();
+#ifdef WARPX_DIM_RZ
+            const int nrz = WarpX::n_rz_azimuthal_modes;
+            const int n_rz_comp = WarpX::ncomps;
+#endif
             amrex::ParallelFor( tbx, ncomp_dst,
                 [=] AMREX_GPU_DEVICE(int i, int j, int k, int n)
                 {
+                    // Field id that corresponds to the nth user-requested component
                     const int icomp = field_map_ptr[n];
 #if defined(WARPX_DIM_3D)
                     dst_arr(i, j, k_lab, n) = src_arr(i, j, k, icomp);
-#else
+#elif defined(WARPX_DIM_XZ)
                     dst_arr(i, k_lab, k, n) = src_arr(i, j, k, icomp);
+#elif defined(WARPX_DIM_RZ)
+                    // rzcomp below gives the component id, 0 to (n_rz_comp-1) for a given field
+                    const int rzcomp = n - static_cast<int>( floor(n/(2.0*nrz-1.0)) ) * n_rz_comp;
```

Comment:
Hm, could this simply be written as a modulo operation?
```suggestion
                    const int rzcomp = n % n_rz_comp;
```

Reply:
This is the same. 
Does modulo affect the speed in the GPU code, though?